### PR TITLE
FUSETOOLS-3186 - avoid usage of slow IFile.getContentDescription and

### PR DESCRIPTION
**Pull request by apupier**
filter for xml files

it is trying to synchronize too many things for our case. it is faster
to reparse the file and check the content ourself

Signed-off-by: Aurélien Pupier <apupier@redhat.com>

# Pull Request Checklist

After this checklist is all checked or PR provides explanations for possible pass-through, please put the label "Ready for review" on the PR.

## General

- [x] Did you use the Jira Issue number in the commit comments?
- [x] Did you set meaningful commit comments on each commit?
- [x] Did you sign off all commits for this PR? (git commit -s -m "jira issue number - your commit comment")

## Functional

- [x] Did the CI job report a successful build?
- [x] Is the non-happy path working, too?

## Maintainability

- [x] Are all Sonar reported issues fixed or can they be ignored?
- [x] Is there no duplicated code?
- [x] Are method-/class-/variable-names meaningful?

## Tests

- [x] Are there unit-tests?
- [x] Are there integration tests (or at least a jira to tackle these)?
- [ ] Do we need a new UI test?

## Legal

- [x] Have you used the correct file header copyright comment?
- [x] Have you used the correct year in the headers of new files?

### CODE REVIEW

**Comment by apupier** [comment]
before
image

after
image

from 1.7s to 94ms (time divided by 18)

File: core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/CamelFilesFinder.java [hunk 1]
```diff
@@ -107,27 +107,34 @@ private boolean isTestProjectFolder(IProject project, IResource f) {
...
 			}
 		}
 		return false;
 	}
+
+	private ByteArrayInputStream createMarkableAndResetableInputStream(IFile ifile) throws IOException {
+		return new ByteArrayInputStream(Files.readAllBytes(ifile.getLocation().toFile().toPath()));
+	}
+
+	private String getContentFileID(IContentDescription contentDescription) {
```
Comment (bfitzpat) [inline core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/CamelFilesFinder.java]: Where is this used?

File: core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/CamelFilesFinder.java [hunk 2]
```diff
@@ -107,27 +107,34 @@ private boolean isTestProjectFolder(IProject project, IResource f) {
...
+		return contentDescription.getContentType().getId();
+	}
+
+	private IContentDescription getFileContentDescription(IFile ifile) throws CoreException {
```
Comment (bfitzpat) [inline core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/CamelFilesFinder.java]: Where is this used?

File: core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/CamelFilesFinder.java (hunk 1 above)
Comment (apupier) [reply core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/CamelFilesFinder.java]: it is no more used. Good catch

File: core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/CamelFilesFinder.java (hunk 2 above)
Comment (apupier) [reply core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/CamelFilesFinder.java]: it is no more used. Good catch

**Pull request merged**